### DISCUSSION TITLE
Training_OpenAI_GPT_2.ipynb for tensorflow 2.x

### DISCUSSION THREAD

**Discussion by lxt3** [Denis2054/Transformers-for-NLP-2nd-Edition#8] [General]
In case anyone is interested, I think I was able to get Training_OpenAI_GPT_2.ipynb working in tensorflow 2.0 (version 2.6, actually). At least it appears so. I didn't use colab, but my own system.

In addition to liberal use of tf.compat.v1, I got rid of tf.summary,merge, and used namedtuple for the get_hparams function. I did not need to modify download_model.py, encoder.py, load_dataset.py, or accumulate.py.

I ignored deprecation warnings, though. By the way, chatGPT/Bard failed at almost any suggestion to translate to tensorflow 2.0. It couldn't think past compat.v1, and often suggested tensorflow 1.0 code.

Perhaps outdated, but I like that we start with py source files, even though I don't really know python well at all (which could mean my fixes are not correct...).

**Comment by Denis2054** [comment]
Good job! Thank you for pointing this out. 
True. If an environment is created with former versions of the libraries the program will work.
Exploring the original GPT-2 code  can be interesting for vintage educational purposes.
